addons/statustimers/main_ui.lua

```diff
@@ -405,6 +406,41 @@ end
 -------------------------------------------------------------------------------
 local module = {};
 
+-- get the current menu target index.
+--
+-- NOTE: I assume this is an index, since most menus start at 1 and increment,
+-- but not all are in order and some are missing indices. For the status icons
+-- menu, it seems to stay in order.
+local function get_current_menu_target()
+    local pointer = ashita.memory.find('FFXiMain.dll', 0, "8B510885D274183B05????????7509", 9, 0)
+    local subPointer = ashita.memory.read_uint32(pointer)
+
+    local subValue = ashita.memory.read_uint32(subPointer)
+    if (subValue == 0) then
+        return ''
+    end
+
+    return ashita.memory.read_uint32(subValue + 0x4C)
+end
+
+-- get the current menu name
+local function get_menu_name()
+    local pGameMenu = ashita.memory.find('FFXiMain.dll', 0, "8B480C85C974??8B510885D274??3B05", 16, 0)

```

Comment:
`"8B480C85C974??8B510885D274??3B05"` and `"8B510885D274183B05????????7509"` will find the same same struct - you can get the menu index from the same pointer as the menu name by using the correct offset.

<pre>
--------------|8B510885D274183B05|????????7509 
8B480C85C974??|8B510885D274??3B05|--------------
</pre>

See the overlap?

Reply:
That was a good catch. I went with the signature used for the menu name, instead of the one I used for index, since other addons also use it.